src/memote.jl

```diff
@@ -321,3 +321,73 @@ function is_consistent(
     value.(m)
     termination_status(opt_model) == OPTIMAL
 end
+
+
+
+"""
+Checks for the presence of metabolite annotations.
+Returns a vector of all metabolites without any annotations.
+"""
+
+function all_unannotated_metabolites(model::StandardModel)
+    return [metabolite_id for metabolite_id in metabolites(model) if isempty(metabolite_annotations(model, metabolite_id))]
+end
+
+
+
+"""
+Iterates through a model's metabolites to check if any common biochemical databases appear in the annotations field.
+Returns a dictionary with the biochemical database names being checked for as keywords 
+and the corresponding value being a vector of all metabolites missing that database name.
+"""
+
+function unannotated_metabolites(model::StandardModel,
+    annotation_keywords = ["pubchem.compound", "kegg.compound", "seed.compound", "inchi_key", "inchi", 
+    "chebi", "hmdb", "reactome", "metanetx.chemical", "bigg.metabolite", "biocyc"])
+   missing_annos = Dict{String, Vector{String}}()
+   for keyword in annotation_keywords
+       missing_annos[keyword] = []
+       for metabolite_id in metabolites(model)
+           metabolite_annos = metabolite_annotations(model, metabolite_id)
+           !haskey(metabolite_annos, keyword) && push!(missing_annos[keyword], metabolite_id)
+       end
+   end
+   return missing_annos
+end
+
+
+
+"""
+Metabolite Annotation Conformity Per Database
+Uses the following regex dictionary to check if the annotations conform to patterns defined according to the MIRIAM guidelines,
+i.e. matching those that are defined at https://identifiers.org/.
+Returns a dictionary with a key for every database that has annotations 
+with the corresponding value being an array of all metabolites whose annotations do not match the given regex pattern.
+"""
+
+metabolites_regex = Dict("pubchem.compound" => r"^\d+$",
+    "kegg.compound" => r"^C\d+$",
+    "seed.compound" => r"^cpd\d+$",
+    "inchi_key" => r"^[A-Z]{14}\-[A-Z]{10}(\-[A-Z])?",
+    "inchi" => r"^InChI\=1S?\/[A-Za-z0-9\.]+(\+[0-9]+)?(\/[cnpqbtmsih][A-Za-z0-9\-\+\(\)\,\/\?\;\.]+)*$",
```

Comment:
is there please a source for this regex? I'm not super familiar with inchi but it looks like there is a bit of magic. Also, is it just a parsing check or does it check for validity of the inchi string (as in, if it does describe an actual molecule)?

Reply:
It's probably from memote.py. But please convert to a readableregex to make it more readable... See here: https://github.com/jkrumbiegel/ReadableRegex.jl

Reply:
hmm ReadableRegex has no deps... And makes regex statements much more readable...

Reply:
Ah OK, maybe the sourcing comment was just too far for me to see. If there's a big overall reference to Memote.py, let's keep it there and it's good that way I'd say.

ReadableRegex is a currently very unstable dependency so let's prevent it until absolutely necessary, this regex here is pretty much okay :D

Reply:
Also what is the final verdict re: readable regex?

Reply:
not needed, it doesn't really improve the situation here. I'll eventually make an actual parser that checks if the inchi encodes an actual molecule (instd. of just looking like inchi)

Reply:
(that's also the basic reason for my original question -- mostly, it was "how come this regex is valid" :D )